Ported from mumble-developers-qt/4.8-mumble
eb5812113105ddcf751c8d407bcbc70d768cdec1

--- ./qtbase/qmake/generators/makefiledeps.cpp
+++ ./qtbase/qmake/generators/makefiledeps.cpp
@@ -654,7 +654,7 @@ bool QMakeSourceFileInfo::findDeps(Sourc
                 if(QDir::isRelativePath(lfn.real())) {
                     if(try_local) {
                         QDir sourceDir = findFileInfo(sourceFile).dir();
-                        QMakeLocalFileName f(sourceDir.absoluteFilePath(lfn.local()));
+                        QMakeLocalFileName f(sourceDir.relativeFilePath(lfn.local()));
                         if(findFileInfo(f).exists()) {
                             lfn = fixPathForFile(f);
                             exists = true;
